### DISCUSSION TITLE
Is it safe to share a cache?

### DISCUSSION THREAD

**Discussion by LittleFox94** [ccache/ccache#828] [Questions & answers] [answered]
Hi,

can a shared cache be a security problem when shared with untrusted users, i.e. is it possible to introduce malicious code into other peoples builds?

Use case: I have a public gitlab instance with public, non-privileged runner and am thinking of adding a ccache volume anyone can use.

**Comment by jrosdahl** [comment, marked as answer]
Hi,

Since a ccache cache is file-based, sharing a cache with a user means allowing that user to write anything to the cache area (unless you're only allowing the user to read results by making it read-only). It is therefore more or less trivial to overwrite a cached result (object file) with a malicious version. In other words, all users that have write access to the cache have to be trusted.

**Reply by LittleFox94** [reply]
thank you :)
